feat : Added network switch, disabled components for wrong network and completed bridge UI

0xthi opened the pull request:
- Added network switch button, which shows supported networks (Sepolia, Base Sepolia), if connected wrong shows warning
- Disabled components if connected to unsupported network and added warning to switch
- If bridge supported network connected, instead of Deposit modal, bridge modal will open, which is integrated with across
- Shows progress of bridging from Approve to filling

0xthi committed:
across bridge, get quote

0xthi committed:
Across bridge+action component

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into thileep/bac-123-across-integration-for-cross-chain-bridging

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into thileep/bac-123-across-integration-for-cross-chain-bridging

0xthi committed:
refactor: Replaced deposit with PermitDeposit in Across

PermitDeposit implemented instead of normal deposit in acrossBridge.

This change will enable permit2 signing in across cross chain intent

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into ui-225-actual-order-amount-is-not-editable

0xthi committed:
feat: Added network switch and bridge UI

Network switch added, components will be disabled for wrong network and bridge UI completed

It solves the issue that app only works with supported network and bridge networks

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into ui-153-add-conditional-rendering-logic-on-the-app-based-on-network

0xthi committed:
fix: Build error fix

refactored code to solve syntax and dependency error

Build error resolved

Prithvi101 commented on src/components/modals/BridgeComponent.tsx:
can u use base model component for reference you can see add funds popup

Prithvi101 commented on src/app/trade/page.tsx:
```diff
@@ -17,11 +17,13 @@ import OrderSection from "@/components/trade/OrderSection"
     const setTradeTypeDialogOpen = useSetAtom(isTradeTypeDialogOpen)
     const [tradeType] = useAtom(selectedTradeType)
     const setisDepositModalOpen = useSetAtom(depositModalAtom)
+    const isBridgeSupportedNetwork = useBridgeNetwork()
```
you can put this in modal wrapper no need to add it to all seprate pages

Prithvi101 commented on src/app/portfolio/page.tsx:
```diff
@@ -121,7 +122,7 @@ const PortfolioPage = () => {
                                             onClick={() => setIsDepositModalOpen(true)}
                                         >
-                                            Deposit
+                                            {isBridgeSupportedNetwork ? "Bridge Deposit" : "Deposit"}
```
same here u can put it to modal wrapper

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into ui-153-add-conditional-rendering-logic-on-the-app-based-on-network

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into ui-153-add-conditional-rendering-logic-on-the-app-based-on-network

0xthi committed:
feat: added arbitrum mainnet and feature to add chain in metamask

arbitrum one mainnet is supported. Users can add networks directly to metamask if it is not available

It solves wallet switching issues and adds arbitrum mainnet

0xthi committed:
fix: reduced code redundancy issues

Move network states to single file and reduced code redundancy

0xthi committed:
fix: created wrapper modal for deposit button

it solves code redundancy
